model.py

```diff
@@ -6937,6 +6925,26 @@ def catalog_identifier(self, _db, identifier):
             self.catalog.append(identifier)
             _db.commit()
 
+    def works_updated(self, _db):
+        """Returns all of a collection's works that have been updated since the
+        last time the collection was checked"""
+        if self.catalog:
+            catalog_identifier_ids = [identifier.id for identifier in self.catalog]
+
+            # Find the works in the catalog that have been created or updated
+            # since the collection last checked.
+            query = _db.query(Work).join(Work.coverage_records)
+            query = query.join(Work.license_pools).join(Identifier)
+            query = query.filter(Identifier.id.in_(catalog_identifier_ids))
+            if self.last_checked:
+                query = query.filter(
+                    WorkCoverageRecord.timestamp > self.last_checked
+                )
+
+            return query.all()
```

Comment:
Have this return the query itself, rather than every item in the resultset. This will make it easy to use our existing pagination system.